src/openloco/ui/dropdown.h

```diff
@@ -1,11 +1,58 @@
 #include "../graphics/colours.h"
 #include "../localisation/stringmgr.h"
+#include <cstdlib>
+#include <vector>
+
+struct format_arg
+{
+    uint8_t type;
+    union
+    {
+        uint32_t u32;
+        uint16_t u16;
```

Comment:
You ought to unify with two `uint16_t`'s rather than one here, right?

Reply:
Why?

Reply:
As I understand it, each of the data types in a union should be of the same size. Consider what happens if you don't. What part of the `uint32_t` does the `uint16_t` point to?

Reply:
As far as I'm aware, union members don't need to be the same length. It's just that you need to read the same variable as your write to.